visualizer/exampleGenerationLinks.js

```diff
@@ -0,0 +1,298 @@
+/* eslint-env browser */
+
+// TODO: update selectable links on neededExamples
+// TODO: nicer formatting for example display
+// TODO: live-updating example displays
+
+'use strict';
+
+(function(root, initModule) {
+  if (typeof exports === 'object') {
+    module.exports = initModule;
+  } else {
+    initModule(root.ohmEditor, root.exampleWorkerManager,
+               root.cmUtil, root.utils, root.domUtil, root.d3);
+  }
+})(this, function(ohmEditor, exampleWorkerManager, cmUtil, utils, domUtil, d3) {
+  var grammar;
+  var grammarEditor;
+
+  var mouseCoords;
+
+  var mousePositionInfo = {
+    mark: null,
+    rule: null,
+    position: null
+  };
+
+  var clickableMarks;
+
+  var exampleDisplay = {
+    lineWidget: null,
+    DOM: null,
+    rule: null
+  };
+
+  var isMouseDown;
+  var linksEnabled;
+
+  // TODO: spin off common functionality as helpers
+  function isPlatformMac() {
+    return /Mac/.test(navigator.platform);
+  }
+
+  // modifier+Shift
+  function areLinksEnabled(e) {
```

Comment:
It's bit misleading to call this `areLinksEnabled` -- from that name, I would expect this to be a pure function, rather than to assign into a shared variable. I'd suggest `updateLinksEnabled` instead.